Fix e2e tests to run if datastore name has space

ashahi1 opened the pull request:
Fixes issue #1625

Have modified E2E tests such that these tests can work even if datastore name has space.

Ran all E2E tests on datastore have spaces in their names.

govint commented on tests/e2e/restart_test.go:
```diff
@@ -330,7 +330,8 @@ func (s *RestartTestData) TestDuplicateVolumeName(c *C) {
 
 	// Cleanup volume on the second datastore
-	out, err = dockercli.DeleteVolume(s.config.DockerHosts[1], s.volumeName+"@"+ds2)
+	volName := "'" + s.volumeName + "@" + ds2 + "'"
```
Suggest making a util function CreateFQVN() that handles adding the _'_ to create a full volume name with its datastore. So this code isn't repeated. The function returns a string that can be used in any command string.

govint commented on tests/e2e/vmgroups_test.go:
```diff
@@ -66,7 +66,7 @@ func (vg *VmGroupTest) SetUpSuite(c *C) {
 	}
 	// Create the test VM group1
-	cmd = adminconst.CreateVMgroup + vgTestVMgroup1 + " --default-datastore " + vg.config.Datastores[0]
+	cmd = adminconst.CreateVMgroup + vgTestVMgroup1 + " --default-datastore '" + vg.config.Datastores[0] + "'"
```
Will it work if GetTestConfig returns the datastore names with the _'_ added?

govint replied:
This works and I'm assuming that any command line (need to check) done this way for the datastore name _should_ work.

_docker volume inspect snapvol2@'sharedVmfs-0'_
[
    {
        "Driver": "vsphere",
        "Labels": null,
        "Mountpoint": "/mnt/vmdk/snapvol2@sharedVmfs-0",
        "Name": "snapvol2@sharedVmfs-0",
        "Options": {},
        "Scope": "global",
        "Status": {
            "access": "read-write",
            "attach-as": "independent_persistent",
            "capacity": {
                "allocated": "15MB",
                "size": "100MB"
            },
            "clone-from": "None",
            "created": "Sun Aug  6 13:01:34 2017",
            "created by VM": "worker1-VM1.0",
            "datastore": "sharedVmfs-0",
            "diskformat": "thin",
            "fstype": "ext4",
            "status": "detached"
        }
    }
]

govint replied:
So, could try out just changing GetTestConfig() to return the datastore names to prefix-suffix with _'_

govint commented on tests/e2e/restart_test.go:
```diff
@@ -73,7 +74,8 @@ func (s *RestartTestData) SetUpSuite(c *C) {
 	// Create a volume on the second datastore
-	out, err = dockercli.CreateVolume(s.config.DockerHosts[1], s.volumeName+"@"+ds2)
+	ds2 = strings.Replace(ds2, "'", "", -1)
+	out, err = dockercli.CreateVolume(s.config.DockerHosts[1], inputparams.CreateFQVN(s.volumeName, ds2))
```
Is it failing if the command line is formatted with the DS name in "'"

govint replied:
API isn't being used anywhere so the command lines should work with the quoted DS names. Isn't that working?

ashahi1 replied:
config.Datastores already has quotes attached to ds names so adding two back-to-back single quotes fails the command
```
root@sc-rdops-vm02-dhcp-52-237:~# docker volume create -d vsphere --name ''RefcountTest_volume_634502@ shared   Vmfs   - (10)''
-bash: syntax error near unexpected token `('
```
So either we can call CreateFQVN() to add quotes or we can go back to the way it was - s.volumeName+"@"+ds2

govint commented on tests/e2e/restart_test.go:
```diff
@@ -89,7 +91,7 @@ func (s *RestartTestData) TearDownSuite(c *C) {
 
 		// Remove access for second datastore from the default vmgroup
-		out, err = adminutil.RemoveDatastoreFromVmgroup(s.config.EsxHost, adminconst.DefaultVMgroup, s.config.Datastores[1])
+		out, err = adminutil.RemoveDatastoreFromVmgroup(s.config.EsxHost, adminconst.DefaultVMgroup, "'"+ds2+"'")
```
Is this adding back the quotes? Do we need to remove it earlier?

ashahi1 replied:
Yes, we remove it in SetUpSuite as we do strings.Compare() there and since config.Datastores has names with quotes whereas status["datastore"] has names without quotes, comparison fails.
```
if strings.Compare(status["datastore"], strings.Replace(s.config.Datastores[0], "'", "", -1)) == 0 {
		ds2 = s.config.Datastores[1]
	}
```

ashahi1 commented:
@govint I have pushed a fresh commit and in that commit, I have removed the changes that were made to the tests. I have modified the test utils accordingly.

govint commented:
Looks good, lets retain the FQVN() api, its better than creating names with quotes in every test.

pshahzeb commented on tests/e2e/vmgroups_test.go:
```diff
@@ -113,7 +113,7 @@ func (vg *VmGroupTest) TearDownSuite(c *C) {
 	cmd := adminconst.ListVMgroups
 	out, err = ssh.InvokeCommand(vg.config.EsxHost, cmd)
-	log.Printf(out)
+	// log.Printf(out)
```
Why to silence these prints?

ashahi1 replied:
This is how it looks like on CI console. I just feel it unnecessarily uses up the log space.
```
2017/08/11 02:12:54 stty: standard input: Inappropriate ioctl for device
Uuid            Name           Description      Default_datas..  VM_list          
--------------  -------------  ---------------  ---------------  ---------------  
11111111-111..  _DEFAULT       This is a def..  _VM_DS                            
e3290eef-591..  vmgroup_test1                   datastore1       photon.vmfs,U..  
ca6e7108-704..  vmgroup_test2                   vsandatastore
```
And we can always enable it for debug purposes.

govint replied:
Ok, I'd added this earlier but with all the tests out there this may be is not needed any more except for debug.

pshahzeb commented on tests/utils/dockercli/containers.go:
```diff
@@ -122,7 +124,10 @@ func GetVolumeStatus(hostName, volumeName string) (map[string]string, error) {
 
 	status := make(map[string]string)
-	val := strings.Fields(out)
+	val := strings.SplitAfter(out, volmPropertyDelimiter)
```
will this not work https://golang.org/pkg/strings/#Split
with sep as volmPropertyDelimiter

pshahzeb commented on tests/utils/esx/govc.go:
```diff
@@ -85,12 +85,17 @@ func ShutDownVM(vmName string) string {
 	cmd := esx.DatastoreInfo + esxcliJSON + esx.DatastoreList
 	out, _ := ssh.InvokeCommandLocally(cmd)
-	return strings.Fields(out)
+	val := strings.SplitAfter(out, "\n")
```
Similar comment to use Split instead of SplitAfter. With using split, you can also get rid of line 95

pshahzeb commented on tests/utils/verification/volumeproperties.go:
```diff
@@ -145,7 +145,6 @@ func getVolumeStatusHost(name, hostName string) string {
 // vsphere shared driver
 func GetSharedVolumeStatusHost(name, hostName string) string {
 	cmd := dockercli.InspectVolume + " --format \"{{index .Status \\\"Volume Status\\\"}}\" " + name
-	log.Printf("GetSharedVolumeStatusHost: cmd[]", cmd)
```
why remove this print?
